### PR TITLE
Use uMTPrd instead of f_gser for the default gadget

### PR DESCRIPTION
**Pull request by BluemediaGER**
This pull request replaces the gser function of the default gadget with a full MTP implementation ([uMTPrd](https://github.com/viveris/uMTP-Responder)). Some head units expect an MTP device and may not work correctly if the connected USB device does not behave like one.

### CODE REVIEW

File: aa_wireless_dongle/board/common/kernel.config [hunk 1]
```diff
@@ -2,6 +2,9 @@
 CONFIG_USB_CONFIGFS_UEVENT=y
 CONFIG_USB_CONFIGFS_F_ACC=y
 
+# Add FunctionFS for uMTPrd
+USB_FUNCTIONFS=y
```
Comment (nisargjhaveri) [inline aa_wireless_dongle/board/common/kernel.config:6]: Is this really required? `CONFIG_USB_CONFIGFS_F_FS` is already enabled and that should take care of this, no?

Also, even if this is required, doesn't it need to be `CONFIG_USB_FUNCTIONFS`?

**Review by nisargjhaveri — COMMENTED** [review]
Thanks for this! The change looks good logically, though I'd like to give it a try once to understand a couple of things better. I'll try to find some time soon.

File: aa_wireless_dongle/board/common/rootfs_overlay/etc/init.d/S92usb_gadget [hunk 2]
```diff
@@ -47,12 +50,21 @@ start() {
...
 		echo "$PRODUCT" > strings/0x409/product
-		mkdir functions/gser.usb0
+		mkdir functions/ffs.mtp
 		mkdir configs/c.1
 		echo 500 > configs/c.1/MaxPower
-		ln -s functions/gser.usb0 configs/c.1
+		ln -s functions/ffs.mtp configs/c.1
 	fi
 
+	# Mount FunctionFS and start uMTPrd
```
[the remark this answers is not in the document]
Comment (nisargjhaveri) [reply aa_wireless_dongle/board/common/rootfs_overlay/etc/init.d/S92usb_gadget:59]: `mount` needs to be done after we setup the function above. We can split this into three different init scripts, but that can be a different refactor. Never mind.

File: aa_wireless_dongle/board/common/rootfs_overlay/etc/umtprd/umtprd.conf [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+storage "/" "root folder" "rw,locked"
+
+manufacturer "My Own"
+product "AA Wireless Dongle"
+serial "0"
+firmware_version "Rev A"
+
+interface "MTP"
+usb_vendor_id 0x1d6b
+usb_product_id 0x0105
```
[the remark this answers is not in the document]
Comment (nisargjhaveri) [reply aa_wireless_dongle/board/common/rootfs_overlay/etc/umtprd/umtprd.conf:12]: These doesn't seem to have any effect as such. But we can keep these for now until there is more clarity.

File: aa_wireless_dongle/board/common/kernel.config (hunk 1 above)
Comment (nisargjhaveri) [reply aa_wireless_dongle/board/common/kernel.config:6]: Please remove this?

**Review by nisargjhaveri — CHANGES_REQUESTED** [review]
I was able to play around with this and this looks good. Thanks! :)

File: aa_wireless_dongle/board/common/rootfs_overlay/etc/umtprd/umtprd.conf (hunk 3 above)
[the remark this answers is not in the document]
Comment (BluemediaGER) [reply aa_wireless_dongle/board/common/rootfs_overlay/etc/umtprd/umtprd.conf:12]: My guess is that these options are only used for GadgetFS mode. Unfortunately uMTPrd will not start without them.

File: aa_wireless_dongle/board/common/kernel.config (hunk 1 above)
Comment (BluemediaGER) [reply aa_wireless_dongle/board/common/kernel.config]: You're right. Removed it.

**Comment by BluemediaGER** [comment]
Sorry for the delay, I've been quite busy with other stuff. I've made the necessary changes.

**Review by nisargjhaveri — APPROVED** [review]

**Pull request merged**